dpnp/backend/kernels/dpnp_krnl_fft.cpp

```diff
@@ -157,6 +143,47 @@ void dpnp_fft_fft_c(const void* array1_in,
     return;
 }
 
+template <typename _DataType_input, typename _DataType_output>
+void dpnp_fft_fft_c(const void* array1_in,
+                    void* result1,
+                    const long* input_shape,
+                    const long* output_shape,
+                    size_t shape_size,
+                    long axis,
+                    long input_boundarie,
+                    size_t inverse)
+{
+    const size_t result_size = std::accumulate(output_shape, output_shape + shape_size, 1, std::multiplies<size_t>());
+    if (!(result_size && shape_size))
+    {
+        return;
+    }
+
+    if constexpr (std::is_same<_DataType_input, std::complex<double>>::value &&
+                  std::is_same<_DataType_output, std::complex<double>>::value)
+    {
+        cl::sycl::event event;
+        _DataType_input* array_1 = reinterpret_cast<_DataType_input*>(const_cast<void*>(array1_in));
+        _DataType_output* result = reinterpret_cast<_DataType_output*>(result1);
+
+        oneapi::mkl::dft::descriptor<mkl_dft::precision::DOUBLE, mkl_dft::domain::COMPLEX> desc(result_size);
+        desc.set_value(mkl_dft::config_param::BACKWARD_SCALE, (1.0 / result_size));
+        desc.set_value(mkl_dft::config_param::PLACEMENT, DFTI_NOT_INPLACE); // enum value from math library C interface
+        // instead of mkl_dft::config_value::NOT_INPLACE
+        desc.commit(DPNP_QUEUE);
+
+        event = mkl_dft::compute_forward(desc, array_1, result);
+        event.wait();
```

Comment:
Please move this function into `dpnp_fft_fft_mathlib_c`

Reply:
PR in progress.